### ISSUE TITLE
Hide / Show properties and default values

### ISSUE THREAD

**Issue by jma** [ngx-formly/ngx-formly#2439]
**Is your feature request related to a problem? Please describe.**
I would be great that the model is in sync with the hidden or shown properties else the submission can be disabled due to an hidden property.

**Describe the solution you'd like**

1. If a default value exists for a hidden property it should not be present in the model / validation process.
1. When property becomes visible the default value should be restored.

**Describe alternatives you've considered**
A global configuration can be proposed to choose the desired behaviour.

**Additional context**

Here is a simple example:
https://stackblitz.com/edit/formly-hide-show-default

**Comment by aitboudad** [comment]
I've pushed a dev release `5.10.0-dev.ad26817d` where you may use until `5.10.0` (still not have the time :pray:), so I've introduced a new config which takes account of default value on reset, so please give it a try and let me know if that fixed the issue on your side:


```patch
    FormlyModule.forRoot({
+      extras: {
+        resetFieldOnHide: true,
+      },
    }),
```

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
